### PR TITLE
Expand lint scripts

### PR DESCRIPTION
**Pull request by LeviPesin**
Related issue: -

**Description**

Expand lint test to include tests for addons, docs, and other directories. This PR also changes main `lint` script so these tests are ran by `npm run lint` (and therefore the lint test in the actions).

### CODE REVIEW

**Comment by LeviPesin** [comment]
CodeQL alerts on this PR because linting changes in some files trigger it (CodeQL thoughts that issues were both removed and added).

**Comment by LeviPesin** [comment]
The number of lines changed is scary 😅

**Comment by LeviPesin** [comment]
@mrdoob @Mugen87 What do you think about this PR? I think it is good to have lint test to warn if incorrect linting appears in addons (docs, etc).

**Comment by gkjohnson** [comment]
I recommend making smaller PRs with a smaller sets of files linted per PR so the change set is more approachable. And then a separate PR with an automated lint process added.

The `/manual` directory has never matched the project code style since it was developed in another source - so I expect that to require a large set of changes.

**Comment by Mugen87** [comment]
Agreed! It's always hard to review so large PRs. It's easy to overlook something so smaller change sets are preferable.

I would exclude linting `manual` for now. It's not only the code style which is somewhat out of line comparing the examples.

**Comment by LeviPesin** [comment]
Reverted the lint changes to manual. Can this PR be reopen, please?

**Comment by Mugen87** [comment]
I think it would be also better to revert the changes to `package.json`. That should also be discussed in a separate PR.

**Comment by LeviPesin** [comment]
Done.

File: package.json [hunk 1]
```diff
@@ -97,10 +102,16 @@
...
+    "lint-core": "eslint src",
+    "lint-addons": "eslint examples/jsm --ext .js --ignore-pattern libs --ignore-pattern ifc",
+    "lint-examples": "eslint examples --ext .html",
+    "lint-docs": "eslint docs --ignore-pattern prettify.js",
+    "lint-editor": "eslint editor --ignore-pattern libs",
+    "lint-manual": "eslint manual --ignore-pattern 3rdparty --ignore-pattern prettify.js --ignore-pattern shapefile.js",
+    "lint-test": "eslint test --ignore-pattern vendor",
+    "lint-utils": "eslint utils",
+    "lint": "npm run lint-core",
+    "lint-fix": "npm run lint-core -- --fix",
```
Comment (Mugen87) [inline package.json:114]: Please undo the changes to the `scripts` section.

Comment (LeviPesin) [reply package.json:114]: This just adds new scripts for linting the corresponding parts, this does not change the main lint test anymore (it tests only core as it was before).

Comment (Mugen87) [reply package.json]: >  this does not change the main lint test anymore

That's good since I don't think we should change the semantics of `npm run lint`.

File: test/benchmark/core/Vector3Components.js [hunk 2]
```diff
@@ -108,6 +108,8 @@
 
 		}
 
+		return result;
```
Comment (Mugen87) [inline test/benchmark/core/Vector3Components.js:111]: Why is it necessary to return the result in the `add()` methods?

Comment (LeviPesin) [reply test/benchmark/core/Vector3Components.js:111]: Otherwise ESLint complains that `result` variable is unused.

**Review by Mugen87 — APPROVED** [review]
